---
title: "Practica 4"
author: "J. Rojas, X. Díaz"
date: "3/7/2022"
output: 
  html_notebook:
    toc: yes
  html_document:
    toc: yes
  pdf_document:
    toc: yes
  word_document:
    toc: yes
editor_options: 
  markdown: 
    wrap: 72
---

# Repository

Git hub project: <https://github.com/XochitlDiaz/BioifoII_pubrep>

<h4>Save R history</h4>
history(max.show = Inf, reverse = T)




# Cluster Usage

```{r, eval=FALSE}
module load kallisto/0.45.0
module load sra/2.9.6-1
module load r/4.0.2

```

# Data Description

Data was obtained by using the sra package and the respective accesion list found in it's GEO Dataset. Our SRA Accession is: GSE185492. This was found by running a search in th NCBI Databse and then sending the selected files to the SRA Run Selector to generate an accession list.

The data itself was generated by RNA-seq of control and treated mice. The goal of this study was to the investigate the hepatic effects of lactational deliver of Triclosan in neonatal mice. The treatment group was Triclosan treated mice via milk ingest and it's respective control group (N=3/group).

In this study, DE analysis was performed to identify the transcriptomic profile of the control and treatment group.


## Data Download

<h3> Commands </h3>

    scp <storage_dir>/SRR_Acc_List.txt crojas@dna.lavis.unam.mx:´<cluster_dir>/SRR_Acc_List.txt
    xargs fastq-dump <SRR_Acc_List.txt
    scp <storage_dir>/SraRunTable.txt crojas@dna.lavis.unam.mx:´<cluster_dir>/SraRunTable.txt

<h3> Explanation </h3>

To download the data, we first generated an accesion list containing only the replicas used in this report. The aforementioned accesion list was then uploaded to our server using the scp command.
Finally, to download all the files contained in the accession list, the fastq-dump command was used. To run the command for every file contained in our accession list, the expression: "<SRR_Acc_List.txt" was employed.
We also downloaded the associated metadata, since it will come in handy both for understanding the treatment status of each sample and to tag each simple in our differential expression analysis.

# RNA-seq Pipeline


## Quality Control

After checking the metadata and data description of our reads, we realized that they had already been filtered and set to the same length. Therefore, it was no longer necessary to perform any clipping or filtering.

## Indexing

### Download Reference Transcriptome

<h3>

Commands

</h3>

```{r, eval=FALSE}
wget https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M29/gencode.vM29.transcripts.fa.gz
```

<h3>

Explanation

</h3>

The wget command allows us to download a file by providing the file URL
from a public available server. We downloaded the full Mouse transcriptome from the GENCODE webpage.

## Kallisto Indexing

<h3>
Commands
</h3>

```{r, eval=FALSE}
kallisto index -i index_gencode_v39_transcripts gencode.v39.transcripts.fa.gz
```

<h3>
Explanation
</h3>

This command helps us index our transcriptome reference file. The -i
parameter is used to determine the output of our indexed transcriptome. This will allow us to perform kallisto quantification much faster and efficiently.

## Quantification

### Obtain Paramaters

<h3> Commands </h3>

```{r, eval=FALSE}
awk 'BEGIN { t=0.0;sq=0.0; n=0;} ;NR%4==2 {n++;L=length($0);t+=L;sq+=L*L;}END{m=t/n;printf("total %d avg=%f stddev=%f\n",n,m,sq/n-m*m);}' SRR18735738.fastq
awk 'BEGIN { t=0.0;sq=0.0; n=0;} ;NR%4==2 {n++;L=length($0);t+=L;sq+=L*L;}END{m=t/n;printf("total %d avg=%f stddev=%f\n",n,m,sq/n-m*m);}' SRR18735739.fastq
awk 'BEGIN { t=0.0;sq=0.0; n=0;} ;NR%4==2 {n++;L=length($0);t+=L;sq+=L*L;}END{m=t/n;printf("total %d avg=%f stddev=%f\n",n,m,sq/n-m*m);}' SRR18735740.fastq
awk 'BEGIN { t=0.0;sq=0.0; n=0;} ;NR%4==2 {n++;L=length($0);t+=L;sq+=L*L;}END{m=t/n;printf("total %d avg=%f stddev=%f\n",n,m,sq/n-m*m);}' SRR18735741.fastq
awk 'BEGIN { t=0.0;sq=0.0; n=0;} ;NR%4==2 {n++;L=length($0);t+=L;sq+=L*L;}END{m=t/n;printf("total %d avg=%f stddev=%f\n",n,m,sq/n-m*m);}' SRR18735742.fastq
awk 'BEGIN { t=0.0;sq=0.0; n=0;} ;NR%4==2 {n++;L=length($0);t+=L;sq+=L*L;}END{m=t/n;printf("total %d avg=%f stddev=%f\n",n,m,sq/n-m*m);}' SRR18735743.fastq
```

<h3>
Explanation
</h3>

Since our data is presented as single-read ends, we must specify the
average length and standard deviation for each of our sequenced group of
reads. To do this we used the regular awk command to obtain the
aforementioned information. Then we used this output to run quantification
with kallisto.

### Kallisto Quantification

<h3>
Commands
</h3>

```{r,eval=FALSE}
mkdir <each sample name>
kallisto quant -i /mnt/Timina/bioinfoII/crojas/RNA/gencode_index -o ./SRR18735738 ../sra/SRR18735738.fastq --single -l 100.00000 -s 0.00001
kallisto quant -i /mnt/Timina/bioinfoII/crojas/RNA/gencode_index -o ./SRR18735739 ../sra/SRR18735739.fastq --single -l 100.00000 -s 0.00001
kallisto quant -i /mnt/Timina/bioinfoII/crojas/RNA/gencode_index -o ./SRR18735740 ../sra/SRR18735740.fastq --single -l 100.00000 -s 0.00001
kallisto quant -i /mnt/Timina/bioinfoII/crojas/RNA/gencode_index -o ./SRR18735741 ../sra/SRR18735741.fastq --single -l 100.00000 -s 0.00001
kallisto quant -i /mnt/Timina/bioinfoII/crojas/RNA/gencode_index -o ./SRR18735742 ../sra/SRR18735742.fastq --single -l 100.00000 -s 0.00001
kallisto quant -i /mnt/Timina/bioinfoII/crojas/RNA/gencode_index -o ./SRR18735743 ../sra/SRR18735743.fastq --single -l 100.00000 -s 0.00001
```

<h3>
Explanation
</h3>

Since kallisto quant command produces its output into directories for
each of the test file's name, first we created directories to fit this
requirement, then we ran the specified commands in an sge file.

The parameter (-i) in kallisto quant is used to specify the route and
name of our index files.\
The -o parameter is used to specify the output directory for each of the
fastq files being quantified. Since these are single end reads, we must
specify this with the flag "--single". The -l and -s parameters are used
to specify mean fragment length and standard deviation for each file.
Finally, we provide the command with each of the files we will quantify.

## Normalization

### Batch effect correction

Since all of the samples were sequenced in the same batch, there shouldn't
be any bias driven by variation in the sequencing batches. Therefore, it was not necessary to undergo batch effect correction.

### DESeq2 

Because we want to make differential expression analysis
through RNA composition comparison between the samples of the
experiment, we used DESeq2's normalization, given that it is an
appropriate method for the data we are using.

<h3>
Commands
</h3>

```{r,eval=FALSE}

cd /mnt/Timina/bioinfoII/XJS/Yalbi/kallisto
module load r/4.0.2
R

(In R)
library(tximport)
library(tidyverse)
files <- file.path("/mnt/Timina/bioinfoII/XJS/Yalbi/kallisto", list.dirs(), "abundance.h5") 
names(files) <- str_extract(files,"SRR\\d+")
files = files[2:7]

# Load table with trx id and gene id corrspondence
tx2gene <- read.csv("/mnt/Timina/bioinfoII/rnaseq/resources/gencode/gencode.vM28.basic.trx_id-gene_id-no_ver.tsv",stringsAsFactors = F)
# Load table with trx id and gene name correspondence
tx2genename <- read.csv("/mnt/Timina/bioinfoII/rnaseq/resources/gencode/gencode.vM28.basic.trx_id-gene_name-no_ver.tsv",stringsAsFactors = F)

txi.kallisto <- tximport(files, type = "kallisto", tx2gene = tx2gene, ignoreAfterBar=TRUE, ignoreTxVersion=TRUE)
txi.kallisto.name <- tximport(files, type = "kallisto", tx2gene = tx2genename, ignoreAfterBar=TRUE, ignoreTxVersion=TRUE)

> library(DESeq2)
samples <- read.csv("/mnt/Timina/bioinfoII/XJS/Yalbi/SraRunTable.txt",stringsAsFactors = F, header = TRUE)
samples <- column_to_rownames(samples, var = "Run")

dds <- DESeqDataSetFromTximport(txi.kallisto,colData = samples,design = ~ treatment)

keep <- rowSums(counts(dds)) >= 6
dds <- dds[keep, ]
dds$treatment <- factor(dds$treatment, levels = c("TCS breastmilk","Regular breastmilk"))
dds <- DESeq(dds)
res <- results(dds)
```

<h3>
Explanation
</h3>

First we loaded R 4.0.2 in our working environment, then we opened R
form the cluster terminal to run the code shown above. It is important
to mention that to run the code we generated the R working environment
at the working directory where we saved the output of kallisto
quantification.

First we used tximport to integrate the transcript data of our 6
samples in a matrix with Mus Musculus genes. We used version 39 of the
genome for this data integration.

To integrate the metadata of our samples in order to make indicate
sample groups (condition groups) we loaded the metadata file onto our R environment and then extracted the appropiate columns that associate treatment group with sample name.


For the DESeq2 Data Set construction we use our tximport output matrix
and indicate that that condition column has the group variables. After
the the statistical modeling we filtered genes that had counts lower or
equal to zero. Then we apply the Differential expression analysis with
DESeq() function and finally we apply the results() function to see the
log 2 fold change analysis in both groups (TCS breastmilk and Regular breastmilk).


## Reducing significant DE genes

<h3> Commands </h3>

```{r}

de <- as.data.frame(res)
de$diffexpressed <- "NO"
de$diffexpressed[de$log2FoldChange > 1 & de$pvalue < 0.01] <- "UP"
de$diffexpressed[de$log2FoldChange < -1 & de$pvalue < 0.01] <- "DOWN"

library(ggrepel)
de$names <- NA
filter <- which(de$diffexpressed != "NO" & de$padj < 0.01 & (de$log2FoldChange > 1 | de$log2FoldChange < -1)) #<<
de$names[filter] <- rownames(de)[filter] #<<

res.05 <- results(dds, alpha = 0.05)
table(res.05$padj < 0.05)

resLFC1 <- results(dds, lfcThreshold=1)
table(resLFC1$padj < 0.1)

```


<h3> Explanation </h3>

Since the DE Analysis produced a large collection of differentially expressed genes, we decided to narrow our search in order to find a more informative subset of genes. To do this, we added another column in our matrix to classify genes according to whether we consider them as differentially expressed or not. 
Then, we conducted filtering in two ways: 

1. By assesing log fold change 
  + We set our new filtered selection to admit genes with a higher absolute value than 1
2. By setting a lower adjusted p-value threshold
  + Genes with an adj. p-value lower than 0.05
  + Genes with an adj. p-value lower than 0.01 (This became our final collection)

Once we had these subsets, we realized that the 0.01 p-value gave the most direct and concrete results, but we also decided to produce results for other differentialy expressed genes without such a strict p-value threshold.


# Results

## Volcano plot

![](./Volcano_plot.png) Shows the overall differentially expressed genes
by comparing the TCS to regular breastmilk. 
![](./Volcano_filt.png) Shows only the genes that passed our LCF1 filter with a 0.01 p-value threshold.
Overexpressed genes are shown in red. Underexpressed genes are
show in blue. Genes that where not significantly over or under expressed
are shown in black.

The most differentially expressed genes are labeled. It is important to see the differences derived from using the full collection of differentially expressed genes versus the filtered subset. In this case, we can observe that the filtered plot seems to have a more uniform distribution, opposed to the obvious outliers seen in the unfiltered collection. Our group of interest is also reduced significantly when taking into account the filtered collection, and since this also means there will be a smaller chance of calling false positives, we will focus on this data.
In corresponding literature we found that these genes are involved in metabolic pathways. 
Specifically, we can see that ENSMUSG00000091144 stands out as an outlier, and in this case this is a protein-coding gene for a zinc finger. To further explore the implications of the overexpressed genes, we will analyze the number of differentially expressed genes by grouping them into function associated families. This will be our functional analysis.


<h4>
Code
</h4>

```{r, eval=FALSE}

# Generate and save normal plot
#save plot as png
outdir = "/mnt/Timina/bioinfoII/XJS/Yalbi/kallisto/Results"
png(file = paste0(outdir,"volcano_filter.png"), width = 800, height = 800) 
ggplot(data=de, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=names)) + #<<
    geom_point() +
  geom_point(x=0.305969,y=1.337297875,col="green") +
    scale_color_manual(values=c("blue", "black", "red")) + #change dot colour #<<
    theme_minimal() +
    geom_text_repel() + #<<
    xlim(-15,15)

dev.off()
```


<h4>Filtered plot</h4>

```{r, eval=FALSE}

png(file = paste0(outdir,"volcano_filt.png"),
    width = 800, height = 800)    
ggplot(data=deLFC1, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=names)) + #<<
    geom_point() +
  geom_point(x=0.305969,y=1.337297875,col="green") +
    scale_color_manual(values=c("blue", "black", "red")) + #change dot colour #<<
    theme_minimal() +
    geom_text_repel() + #<<
    xlim(-15,15)

dev.off()

```




## Functional Analysis

We used Gprofilter to make a gene ontology enrichment analysis to
assess whether the results of our differential expression analysis
made sense or not.


<h4>
Code
</h4>

```{r,eval=FALSE}
library(ggrepel)


library(gprofiler2)
resSig <- subset(res, padj < 0.1 & log2FoldChange > 1)
resSig <- resSig[ order(resSig$log2FoldChange, decreasing = TRUE), ]
DEG <- rownames(resSig)
genes_universe <- rownames(dds)

# enrichment analysis
gostres = gost(query = DEG, organism = "mmusculus", significant = T, 
               correction_method = "fdr", domain_scope = "custom", 
               custom_bg = genes_universe, ordered_query = TRUE)
names(gostres)
attributes(gostres$meta)
head(gostres$result)


library(forcats)
outdir = "/mnt/Timina/bioinfoII/XJS/Yalbi/kallisto/Results/"
go <- as.data.frame(gostres$result)

png(file = paste0(outdir,"gost01-resSig.png"),
    width = 800, height = 800) # guardar el plot en formato png
# Reorder following the value of another column:
library(ggplot2)
library(dplyr)
go %>%
  dplyr::slice(1:20) %>%
  select(p_value, term_name) %>%
  mutate(go = fct_reorder(term_name, p_value)) %>%
  ggplot( aes(x=go, y=p_value)) +
    geom_bar(stat="identity", fill="#3A68AE", alpha=.6, width=.4) +
    coord_flip() +
    labs(x = "", y = "Adjusted P-value") +
    theme_bw(base_size = 20)

dev.off()

```

# References

* Geo Dataset: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE185492

* SRA Run Selector: https://www.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA826106&o=acc_s%3Aa

Ref1: Mónica Padilla 2022, *RNA-seq Bioinformática - LCG-EJ*, Git Hub,
viewed 10 March 2022,
<https://github.com/mpadilla905/clase_RNA-seq_LCGEJ2022.git>

Ref2: EnsEMBL, *Gene: Phf11c ENSMUSG00000091144*, EnsEMBL Core,
<https://www.ensembl.org/Mus_musculus/Gene/Splice?db=core;g=ENSMUSG00000091144;r=14:59618282-59632830;t=ENSMUST00000166912>

Ref3: Pacher Lab, *Kallisto -Maual*, Pacher Lab,
<https://pachterlab.github.io/kallisto/manual>

Ref4: USADELLAB, *Trimmomatic: A flexible read trimming tool for
Illumina NGS data*, USADELLAB ORG,
<http://www.usadellab.org/cms/?page=trimmomatic>

Ref5: MultiQC, *Using MultiQC*, MultiQC Docs,
<https://multiqc.info/docs/>

Ref6: Bioconductor, *Package 'tximport'*, Bioconductor,
<https://bioconductor.org/packages/devel/bioc/manuals/tximport/man/tximport.pdf>

Ref7: Documentation, *MultiQC*,
<https://multiqc.info/docs/>

Ref8: SRA-documentation, *Prefetch to local files*,
<https://github.com/ncbi/sra-tools/issues/71>

