---
title: "Practica 4"
author: "J. Rojas, X. Díaz"
date: "3/7/2022"
output: 
  html_document:
    toc: yes
  pdf_document:
    toc: yes
  word_document:
    toc: yes
editor_options: 
  markdown: 
    wrap: 72
---

Git hub project: <https://github.com/XochitlDiaz/BioifoII_pubrep>

#Save Rhistory
history(max.show = Inf, reverse = T)




# Cluster Usage

```{r, eval=FALSE}
module load kallisto/0.45.0
module load sra/2.9.6-1
module load r/4.0.2

```

# Data Description

Data was obtained by using the sra package and the respective accesion list found in it's GEO Dataset. Our SRA Accession is: GSE185492. This was found by running a search in th NCBI Databse and then sending the selected files to the SRA Run Selector to generate an accession list.

The data itself was generated by RNA-seq of control and treated mice. The goal of this study was to the investigate the hepatic effects of lactational deliver of Triclosan in neonatal mice. The treatment group was Triclosan treated mice via milk ingest and it's respective control group (N=3/group).

In this study, DE analysis was performed to identify the transcriptomic profile of the control and treatment group.

The commands used to retrieve the data was:
    
    scp <storage_dir>/SRR_Acc_List.txt crojas@dna.lavis.unam.mx:´<cluster_dir>/SRR_Acc_List.txt
    xargs fastq-dump <SRR_Acc_List.txt



```{bash, eval=FALSE}
cd /mnt/Timina/bioinfoII/XJS/Yalbi/kallisto
module load r/4.0.2
R

(In R)
> library(tximport)
> library(tidyverse)
> files <- file.path("/mnt/Timina/bioinfoII/XJS/Yalbi/kallisto", list.dirs(), "abundance.h5") 
> names(files) <- str_extract(files,"SRR\\d+")
> files = files[2:7]


# Load table with trx id and gene id corrspondence
tx2gene <- read.csv("/mnt/Timina/bioinfoII/rnaseq/resources/gencode/gencode.vM28.basic.trx_id-gene_id-no_ver.tsv",stringsAsFactors = F)
# Load table with trx id and gene name correspondence
tx2genename <- read.csv("/mnt/Timina/bioinfoII/rnaseq/resources/gencode/gencode.vM28.basic.trx_id-gene_name-no_ver.tsv",stringsAsFactors = F)

txi.kallisto <- tximport(files, type = "kallisto", tx2gene = tx2gene, ignoreAfterBar=TRUE, ignoreTxVersion=TRUE)
txi.kallisto.name <- tximport(files, type = "kallisto", tx2gene = tx2genename, ignoreAfterBar=TRUE, ignoreTxVersion=TRUE)

> library(DESeq2)
samples <- read.csv("/mnt/Timina/bioinfoII/XJS/Yalbi/SraRunTable.txt",stringsAsFactors = F, header = TRUE)
samples <- column_to_rownames(samples, var = "Run")

dds <- DESeqDataSetFromTximport(txi.kallisto,colData = samples,design = ~ treatment)
#dds$treatment <- factor(dds$treatment, levels = c("TCS breastmilk","Regular breastmilk"))

keep <- rowSums(counts(dds)) >= 6
dds <- dds[keep, ]
dds$treatment <- factor(dds$treatment, levels = c("TCS breastmilk","Regular breastmilk"))
dds <- DESeq(dds)
res <- results(dds)

de <- as.data.frame(res)
de$diffexpressed <- "NO"
de$diffexpressed[de$log2FoldChange > 1 & de$pvalue < 0.01] <- "UP"
de$diffexpressed[de$log2FoldChange < -1 & de$pvalue < 0.01] <- "DOWN"

library(ggrepel)
de$names <- NA
filter <- which(de$diffexpressed != "NO" & de$padj < 0.01 & (de$log2FoldChange > 1 | de$log2FoldChange < -1)) #<<
de$names[filter] <- rownames(de)[filter] #<<

res.05 <- results(dds, alpha = 0.05)
table(res.05$padj < 0.05)

resLFC1 <- results(dds, lfcThreshold=1)
table(resLFC1$padj < 0.1)



outdir = "/mnt/Timina/bioinfoII/XJS/Yalbi/kallisto/Results/"
png(file = paste0(outdir,"volcano_plot.png"),
    width = 800, height = 800) #

ggplot(data=de, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=names)) + #<<
    geom_point() +
  geom_point(x=0.305969,y=1.337297875,col="green") +
    scale_color_manual(values=c("blue", "black", "red")) + #change dot colour #<<
    theme_minimal() +
    geom_text_repel() + #<<
    xlim(-15,15)


## For filtered data
png(file = paste0(outdir,"volcano_filt.png"),
    width = 800, height = 800)    
ggplot(data=deLFC1, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=names)) + #<<
    geom_point() +
  geom_point(x=0.305969,y=1.337297875,col="green") +
    scale_color_manual(values=c("blue", "black", "red")) + #change dot colour #<<
    theme_minimal() +
    geom_text_repel() + #<<
    xlim(-15,15)

dev.off()


library(gprofiler2)
resSig <- subset(res, padj < 0.1 & log2FoldChange > 1)
resSig <- resSig[ order(resSig$log2FoldChange, decreasing = TRUE), ]
DEG <- rownames(resSig)
genes_universe <- rownames(dds)

# enrichment analysis
gostres = gost(query = DEG, organism = "mmusculus", significant = T, 
               correction_method = "fdr", domain_scope = "custom", 
               custom_bg = genes_universe, ordered_query = TRUE)
names(gostres)
attributes(gostres$meta)
head(gostres$result)


library(forcats)
outdir = "/mnt/Timina/bioinfoII/XJS/Yalbi/kallisto/Results/"
go <- as.data.frame(gostres$result)

png(file = paste0(outdir,"gost01-resSig.png"),
    width = 800, height = 800) # guardar el plot en formato png
# Reorder following the value of another column:
library(ggplot2)
library(dplyr)
go %>%
  dplyr::slice(1:20) %>%
  select(p_value, term_name) %>%
  mutate(go = fct_reorder(term_name, p_value)) %>%
  ggplot( aes(x=go, y=p_value)) +
    geom_bar(stat="identity", fill="#3A68AE", alpha=.6, width=.4) +
    coord_flip() +
    labs(x = "", y = "Adjusted P-value") +
    theme_bw(base_size = 20)

dev.off()

```

```{r,eval=F}
####More filters

#Lower the false discovery rate threshold (the threshold on `padj` in the results table)


res.05 <- results(dds, alpha = 0.05)
table(res.05$padj < 0.05) # genes that rejected null hypothesis
summary(res.05)
# Raise the log2 fold change threshold from 0 using the `lfcThreshold` argument of results
# test for genes that show more substantial changes due to treatment
# `lfcThreshold = 1`, we test for genes that show significant effects of treatment on gene counts more than doubling or less than halving, because 2^1=2

resLFC1 <- results(dds, lfcThreshold=1)
table(resLFC1$padj < 0.1)
```

## Data Download


# RNA-seq Pipeline


## Quality Control

After checking the metadata and data description of our reads, we realized that they had already been filtered and set to the same length. Therefore, it was no longer necessary to perform any clipping or filtering.


## Download Reference Transcriptome

<h3>

Commands

</h3>

```{r, eval=FALSE}
wget https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M29/gencode.vM29.transcripts.fa.gz
```

<h3>

Explanation

</h3>

The wget command allows us to download a file by providing the file URL
from a public available server. We downloaded the full Mouse transcriptome from the GENCODE webpage.

## Kallisto Indexing

<h3>

Commands

</h3>

```{r, eval=FALSE}
kallisto index -i index_gencode_v39_transcripts gencode.v39.transcripts.fa.gz
```

<h3>

Explanation

</h3>

This command helps us index our transcriptome reference file. The -i
parameter is used to determine the output of our indexed transcriptome. This will allow us to perform kallisto quantification much faster and efficiently.

## Quantification

### Obtain Paramaters

<h3>

Commands

</h3>

```{r, eval=FALSE}
awk 'BEGIN { t=0.0;sq=0.0; n=0;} ;NR%4==2 {n++;L=length($0);t+=L;sq+=L*L;}END{m=t/n;printf("total %d avg=%f stddev=%f\n",n,m,sq/n-m*m);}' SRR2919468_trmd_1.fastq
awk 'BEGIN { t=0.0;sq=0.0; n=0;} ;NR%4==2 {n++;L=length($0);t+=L;sq+=L*L;}END{m=t/n;printf("total %d avg=%f stddev=%f\n",n,m,sq/n-m*m);}' SRR2919469_trmd_1.fastq
awk 'BEGIN { t=0.0;sq=0.0; n=0;} ;NR%4==2 {n++;L=length($0);t+=L;sq+=L*L;}END{m=t/n;printf("total %d avg=%f stddev=%f\n",n,m,sq/n-m*m);}' SRR2919470_trmd_1.fastq
awk 'BEGIN { t=0.0;sq=0.0; n=0;} ;NR%4==2 {n++;L=length($0);t+=L;sq+=L*L;}END{m=t/n;printf("total %d avg=%f stddev=%f\n",n,m,sq/n-m*m);}' SRR2919471_trmd_1.fastq
awk 'BEGIN { t=0.0;sq=0.0; n=0;} ;NR%4==2 {n++;L=length($0);t+=L;sq+=L*L;}END{m=t/n;printf("total %d avg=%f stddev=%f\n",n,m,sq/n-m*m);}' SRR2919472_trmd_1.fastq
awk 'BEGIN { t=0.0;sq=0.0; n=0;} ;NR%4==2 {n++;L=length($0);t+=L;sq+=L*L;}END{m=t/n;printf("total %d avg=%f stddev=%f\n",n,m,sq/n-m*m);}' SRR2919473_trmd_1.fastq
```

<h3>

Explanation

</h3>

Since our data is presented as single-read ends, we must specify the
average length and standard deviation for each of our sequenced group of
reads. To do this we used the regular awk command to obtain the
aforementioned information. Then we used this output to run quantification
with kallisto.

### Kallisto Quantification

<h3>

Commands

</h3>

```{r,eval=FALSE}
mkdir <each sample name>
kallisto quant -i Ref/index_gencode_v39_transcripts.fa -o out/kallisto/SRR2919469 --single -l 88.745296 -s 331.963008 trimmed/SRR2919468_trmd_1.fastq
kallisto quant -i Ref/index_gencode_v39_transcripts.fa -o out/kallisto/SRR2919468 --single -l 90.280727 -s 296.307097 trimmed/SRR2919469_trmd_1.fastq
kallisto quant -i Ref/index_gencode_v39_transcripts.fa -o out/kallisto/SRR2919470 --single -l 87.383824 -s 356.991374 trimmed/SRR2919470_trmd_1.fastq
kallisto quant -i Ref/index_gencode_v39_transcripts.fa -o out/kallisto/SRR2919471 --single -l 85.915299 -s 354.540463 trimmed/SRR2919471_trmd_1.fastq
kallisto quant -i Ref/index_gencode_v39_transcripts.fa -o out/kallisto/SRR2919472 --single -l 86.046524 -s 374.079060 trimmed/SRR2919472_trmd_1.fastq
kallisto quant -i Ref/index_gencode_v39_transcripts.fa -o out/kallisto/SRR2919473 --single -l 85.100154 -s 371.832177 trimmed/SRR2919473_trmd_1.fastq
```

<h3>

Explanation

</h3>

Since kallisto quant command produces its output into directories for
each of the test file's name, first we created directories to fit this
requirement, then we ran the specified commands in an sge file.

The parameter (-i) in kallisto quant is used to specify the route and
name of our index files.\
The -o parameter is used to specify the output directory for each of the
fastq files being quantified. Since these are single end reads, we must
specify this with the flag "--single". The -l and -s parameters are used
to specify mean fragment length and standard deviation for each file.
Finally, we provide the command with each of the files we will quantify.

## Normalization

##Batch effect correction Batch effect correction wasn't necessary
because all of the samples ran under the same batch, so there shouldn't
be any any bias driven by variation in the sequencing batches.

##DESeq2 Because we want to make differential expression analysis
through RNA composition comparison between the samples of the
experiment, we used DESeq2's normalization, given that it is an
appropriate method for the data we are using.

<h3>

Commands

</h3>

```{r,eval=FALSE}
library(tximport)
library(tidyverse)

#load sample data
files <- file.path(list.dirs(),"abundance.h5")
files <- files2[5:10]
names(files) <- str_extract(files,"SRR\\d+") 
files

#Load the data of our reference genome
tx2gene <- read.csv("/mnt/Timina/bioinfoII/rnaseq/resources/gencode/gencode.v38.basic.pc.transcripts.enstid_ensgid-nover.csv",stringsAsFactors = F)
tx2genename <- read.csv("/mnt/Timina/bioinfoII/rnaseq/resources/gencode/gencode.v38.basic.pc.transcripts.enstid_genename-nover.csv",stringsAsFactors = F)

#Generates the relation between reference genome and transcript samples 
txi.kallisto <- tximport(files, type = "kallisto", tx2gene = tx2gene, ignoreAfterBar=TRUE, ignoreTxVersion=TRUE)
txi.kallisto.name <- tximport(files, type = "kallisto", tx2gene = tx2genename, ignoreAfterBar=TRUE, ignoreTxVersion=TRUE)

#To see the order of the data we have
names(txi.kallisto)
#Indicates the transcript counts of each gene for each sample
head(txi.kallisto$counts)

#Metadata of our samples
samples <- read.csv("/mnt/Timina/bioinfoII/squinones/RNA-seq/RNA_MONI/SraRunTable_mod_cond_sort.txt",stringsAsFactors = F, header = TRUE)
#Run is the column with the sample ID
samples <- column_to_rownames(samples, var = "Run")

library(DESeq2)

#DESeq2 Data Set construction
#We specify that the column condition has the value of shNS or shRUNX1
#for each of the samples (the variable in the experiment)
dds <- DESeqDataSetFromTximport(txi.kallisto,colData = samples,design = ~ treatment)

#Filtering to dismiss genes with few counts
keep <- rowSums(counts(dds)) >= 6
dds <- dds[keep, ]

#Establish condition column as a factor value
dds$treatment <- factor(dds$treatment, levels=c("TCS breastmilk","Regular breastmilk"))

dds <- DESeq(dds)

res <- results(dds)
```

<h3>

Explanation

</h3>

First we loaded R 4.0.2 in our working environment, then we opened R
form the cluster terminal to run the code shown above. It is important
to mention that to run the code we generated the R working environment
at the working directory where we saved the output of kallisto
quantification.

First we used taximport to integrate the transcript data of our 6
samples in a matrix with Homo Sapiens genes. We used version 38 of the
genome for this data integration.

To integrate the metadata of our samples in order to make indicate
sample groups (condition groups) we downloaded the Metadata file,
available at :
\#<https://www.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA302327&o=acc_s%3Aa>
Because this metadata file included data of ChiP-Seq, RNA-Seq, and other
sample data, we deleted the rows that where not RNA-seq data from the
.txt file. Additionally becuase the .txt file didn't include information
of a column indicating which samples came form a shNA and shRUNX1
experiment, we looked for further information at:
<https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=>[*sampleID*]{.ul} for
each of the RNA-Seq samples and manually added a column in the .txt
metadata file in order to crate the condition column with the according
data.

For the DESeq2 Data Set construction we use our taximport output matrix
and indicate that that condition column has the group variables. After
the the statistical modeling we filtered genes that had counts lower or
equal to zero. Then we apply the Differential expression analysis with
DESeq() function and finally we apply the results() function to see the
log 2 fold change analysis in both groups (shNA vs shRUNX1).

# Results

<h3>

Volcano plot

</h3>

![](./volcano-res.png) Shows the overall differentially expressed genes
in shRUNX1 sample cell cultures using as base line shNS sample cell
cultures. Overexpressed genes are shown in red. Underexpressed genes are
show in blue. Genes that where not significantly over or under expressed
are shown in gray.

The most deferentially expressed genes are labeled. In the literature we
found that these genes are prognostic markers in various types of
cancer. (Ref8:Ref13)

It is especially interesting that ENSG000001456325 is highly
underexpressed given that it is a tumor suppressor. (Ref14)

<h4>

Code to make the plot

</h4>

```{r, eval=FALSE}
# Add a column to the data frame to specify if they are UP- or DOWN- regulated (log2FoldChange respectively positive or negative)
de <- as.data.frame(res)

# add a column of NAs
de$diffexpressed <- "NO"
# if log2Foldchange > 1 and pvalue < 0.01, set as "UP" 
de$diffexpressed[de$log2FoldChange > 1 & de$pvalue < 0.01] <- "UP"
# if log2Foldchange < -1 and pvalue < 0.01, set as "DOWN"
de$diffexpressed[de$log2FoldChange < -1 & de$pvalue < 0.01] <- "DOWN"

#This library avoids lable overlap in plots
library(ggrepel)

# Create a new column "names" to de, that will contain the name of a subset if genes differentially expressed (NA in case they are not)
de$names <- NA
# filter for a subset of interesting genes
filter <- which(de$diffexpressed != "NO" & de$padj < 0.01 & (de$log2FoldChange > 1 | de$log2FoldChange < -1)) #<<
de$names[filter] <- rownames(de)[filter] #<<

####More filters

#Lower the false discovery rate threshold (the threshold on `padj` in the results table)


res.05 <- results(dds, alpha = 0.05)
table(res.05$padj < 0.05) # genes that rejected null hypothesis
summary(res.05)
# Raise the log2 fold change threshold from 0 using the `lfcThreshold` argument of results
# test for genes that show more substantial changes due to treatment
# `lfcThreshold = 1`, we test for genes that show significant effects of treatment on gene counts more than doubling or less than halving, because 2^1=2

resLFC1 <- results(dds, lfcThreshold=1)
table(resLFC1$padj < 0.1)




# Generate and save plot
#save plot as png
outdir = "/mnt/Timina/bioinfoII/XJS/Yalbi/kallisto/Results"
png(file = paste0(outdir,"volcano_filter.png"), width = 800, height = 800) 
ggplot(data=de, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed, label=names)) + #<<
    geom_point() +
  geom_point(x=0.305969,y=1.337297875,col="green") +
    scale_color_manual(values=c("blue", "black", "red")) + #change dot colour #<<
    theme_minimal() +
    geom_text_repel() + #<<
    xlim(-15,15)

dev.off()
```


Filtered plot

```{r, eval=FALSE}
# Add a column to the data frame to specify if they are UP- or DOWN- regulated (log2FoldChange respectively positive or negative)
deLFC1 <- as.data.frame(resLFC1)

# add a column of NAs
de$diffexpressed <- "NO"
# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP" 
de$diffexpressed[de$log2FoldChange > 0.6 & de$pvalue < 0.05] <- "UP"
# if log2Foldchange < -0.6 and pvalue < 0.05, set as "DOWN"
de$diffexpressed[de$log2FoldChange < -0.6 & de$pvalue < 0.05] <- "DOWN"

png(file = paste0(outdir,"volcano04-res.png"),
    width = 800, height = 800) # guardar el plot en formato png

ggplot(data=de, aes(x=log2FoldChange, y=-log10(pvalue), col=diffexpressed)) + # cambiamos col param de aes #<<
  geom_point() + theme_minimal() + 
  geom_vline(xintercept=c(-0.6, 0.6), col="red") +
  geom_hline(yintercept=-log10(0.05), col="red") +
  xlim(-15,15)

dev.off()
```








<h3>

Functional Analysis

</h3>

We used Gprofilter to make a gene ontology enrichment analysis to
evaluate whether the results of our differential expression analysis
make sense or not.

![Go - over expressed](./gost01-resSig.png) Gene ontology enrichment
analysis of overexpressed genes in shRUNX1 samples: From this figure we
see a significant overexpression of H2AX complex II, which makes sense
in the context of RUNX1 oncogenic mutant given that we previously knew
RUNX1 is associated with genome organization. Searching in the
literature we found H2AX is an H2A histone variant group (ref 2) found
in response to DNA damage, so it makes sense finding it overexpressed in
presence of an activated oncogene because these often lead to more
mutations and DNA damage to which the cell would most likely be trying
to repair. To further understand if this overexpression of H2AX is a
direct cause of RUNX1 mutant or if it's a factor that leads to RUNX1
mutant we would have to make a more ditailed analysis of the literature
or even some experiments, which overpass the purpose of this assay.

![Go - underexpressed](./gost02-resSig.png) Gene ontology enrichment
analysis of underexpressed genes in shRUNX1 samples. Searching in the
literature we found that many of the genes that are significantly
under-expressed are related to extra cellular matrix signaling
components, which may explain why the immunity system fails to recognize
the oncogenic cell in order to kill it. Gene - function relation
(Ref3:Ref7)

-   TGFβ1, TGFβ2, and TGFβ3 : play an essential role in cellular
    signaling and ECM remodeling.

-   PLXNA4 : Signaling for cytoskeleton remodeling

-   MMP14 and MMP2 : Breakdown of extracellular matrix.

-   CLDN1 : Cell - cell spatial interactions.

<h4>

Code to make the plots

</h4>

```{r,eval=FALSE}
library(ggrepel)
library(gprofiler2)

# For down-regulated genes

# subset results for genes of interest
resSig <- subset(res, padj < 0.01 & log2FoldChange < -1) # subset
resSig <- resSig[ order(resSig$log2FoldChange, decreasing = TRUE), ]

# define gene lists
DEG <- rownames(resSig)
genes_universe <- rownames(dds)

# enrichment analysis
gostres = gost(query = DEG, organism = "hsapiens", significant = T, 
               correction_method = "fdr", domain_scope = "custom", 
               custom_bg = genes_universe, ordered_query = TRUE)
               
library(forcats)
library(ggplot2)

outdir = "/mnt/Timina/bioinfoII/squinones/RNA-seq/RNA_MONI/"
go <- as.data.frame(gostres$result)

png(file = paste0(outdir,"gost02-resSig.png"),
    width = 1800, height = 800) # guardar el plot en formato png
# Reorder following the value of another column:
go %>%
  dplyr::slice(1:15) %>%
  select(p_value, term_name) %>%
  mutate(go = fct_reorder(term_name, p_value)) %>%
  ggplot( aes(x=go, y=p_value)) +
  geom_bar(stat="identity", fill="#3A68AE", alpha=.6, width=.4) +
  coord_flip() +
  labs(x = "", y = "Adjusted P-value") +
  theme_bw(base_size = 20)

dev.off()

# For up-regulated genes

# subset results for genes of interest
resSig <- subset(res, padj < 0.01 & log2FoldChange > 1) # subset
resSig <- resSig[ order(resSig$log2FoldChange, decreasing = TRUE), ]

# define gene lists
DEG <- rownames(resSig)
genes_universe <- rownames(dds)

# enrichment analysis
gostres = gost(query = DEG, organism = "hsapiens", significant = T, 
               correction_method = "fdr", domain_scope = "custom", 
               custom_bg = genes_universe, ordered_query = TRUE)
               
library(forcats)
library(ggplot2)

outdir = "/mnt/Timina/bioinfoII/squinones/RNA-seq/RNA_MONI/"
go <- as.data.frame(gostres$result)

png(file = paste0(outdir,"gost02-resSig.png"),
    width = 1800, height = 800) # guardar el plot en formato png
# Reorder following the value of another column:
go %>%
  dplyr::slice(1:15) %>%
  select(p_value, term_name) %>%
  mutate(go = fct_reorder(term_name, p_value)) %>%
  ggplot( aes(x=go, y=p_value)) +
  geom_bar(stat="identity", fill="#3A68AE", alpha=.6, width=.4) +
  coord_flip() +
  labs(x = "", y = "Adjusted P-value") +
  theme_bw(base_size = 20)

dev.off()
```

# References

Ref1: Mónica Padilla 2022, *RNA-seq Bioinformática - LCG-EJ*, Git Hub,
viewed 10 March 2022,
<https://github.com/mpadilla905/clase_RNA-seq_LCGEJ2022.git>

Ref2: Uni Prot , *UniProtKB - P16104 (H2AX_HUMAN)* , Uni Prot,
<https://www.uniprot.org/uniprot/P16104>

Ref3:Uni Prot , *UniProtKB - P36897 (TGFR1_HUMAN)* , Uni Prot,
<https://www.uniprot.org/uniprot/P36897>

Ref4:Wikipedia, *Receptores de TGF-beta* , Wikipedia
<https://es.wikipedia.org/wiki/Receptores_de_TGF-beta>

Ref5:Uni Prot , *UniProtKB - Q9HCM2 (PLXA4_HUMAN)* , Uni Prot,
<https://www.uniprot.org/uniprot/Q9HCM2>

Ref5:Uni Prot , *UniProtKB - P50281 (MMP14_HUMAN)* , Uni Prot,
<https://www.uniprot.org/uniprot/P50281>

Ref6:Uni Prot , *UniProtKB - P08253 (MMP2_HUMAN)* , Uni Prot,
<https://www.uniprot.org/uniprot/P08253>

Ref7:Uni Prot , *UniProtKB - O95832 (CLD1_HUMAN)* , Uni Prot,
<https://www.uniprot.org/uniprot/O95832>

Ref8:Uni Prot , *UniProtKB - P16949 (STMN1_HUMAN)* , Uni Prot,
<https://www.uniprot.org/uniprot/P16949>

Ref9:Uni Prot , *UniProtKB - P68431 (H31_HUMAN)* , Uni Prot,
<https://www.uniprot.org/uniprot/P68431>

Ref10:Uni Prot , *UniProtKB - Q9NQW6 (ANLN_HUMAN)* , Uni Prot,
<https://www.uniprot.org/uniprot/Q9NQW6>

Ref11:Uni Prot , *UniProtKB - P06280 (AGAL_HUMAN)* , Uni Prot,
<https://www.uniprot.org/uniprot/P06280>

Ref12:Wikipedia, *HIST1H3C* , Wikipedia
<https://en.wikipedia.org/wiki/HIST1H3C>

Ref13:Uni Prot , *UniProtKB - Q8IZT6 (ASPM_HUMAN)* , Uni Prot,
<https://www.uniprot.org/uniprot/Q8IZT6>

Ref14:Uni Prot , *UniProtKB - Q9NYY3 (PLK2_HUMAN)* , Uni Prot,
<https://www.uniprot.org/uniprot/Q9NYY3>

Ref15: Pacher Lab, *Kallisto -Maual*, Pacher Lab,
<https://pachterlab.github.io/kallisto/manual>

Ref16: USADELLAB, *Trimmomatic: A flexible read trimming tool for
Illumina NGS data*, USADELLAB ORG,
<http://www.usadellab.org/cms/?page=trimmomatic>

Ref17: MultiQC, *Using MultiQC*, MultiQC Docs,
<https://multiqc.info/docs/>

Ref18: Bioconductor, *Package 'taximport'*, Bioconductor,
<https://bioconductor.org/packages/devel/bioc/manuals/tximport/man/tximport.pdf>
